internal/ais/workflow.go

```diff
@@ -0,0 +1,222 @@
+package ais
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/artefactual-sdps/temporal-activities/archivezip"
+	"github.com/artefactual-sdps/temporal-activities/bucketupload"
+	"github.com/artefactual-sdps/temporal-activities/removepaths"
+	"github.com/google/uuid"
+	"go.artefactual.dev/tools/bucket"
+	temporalsdk_api_enums "go.temporal.io/api/enums/v1"
+	temporalsdk_activity "go.temporal.io/sdk/activity"
+	temporalsdk_client "go.temporal.io/sdk/client"
+	temporalsdk_temporal "go.temporal.io/sdk/temporal"
+	temporalsdk_worker "go.temporal.io/sdk/worker"
+	temporalsdk_workflow "go.temporal.io/sdk/workflow"
+)
+
+const WorkflowName = "ais-workflow"
+
+type WorkflowParams struct {
+	AIPUUID uuid.UUID
+	AIPName string
+}
+
+type WorkflowResult struct {
+	Key string
+}
+
+type Workflow struct {
+	workingDir string
+}
+
+func NewWorkflow(workingDir string) *Workflow {
+	return &Workflow{workingDir: workingDir}
+}
+
+func (w *Workflow) Execute(ctx temporalsdk_workflow.Context, params *WorkflowParams) (r *WorkflowResult, e error) {
```

Comment:
I'll make sure all the activities from this workflow are executed under a session. Right now, because we run the preprocessing child workflow in the same worker, which requires to share the filesystem with Enduro, it won't be an issue if the activities are run from different workers. But the same requirement doesn't apply to this workflow so we could create another worker later on and scale it differently.